index.mjs

```diff
@@ -57,8 +56,14 @@ export default function icomesh(order = 4) {
     }
 
     // normalize vertices
-    for (let i = 0; i < vertices.length; i += 3) {
-        const m = 1 / Math.hypot(vertices[i + 0], vertices[i + 1], vertices[i + 2]);
+    for (let i = 0, len = vertices.length; i < len; i += 3) {
+        // const m = 1 / Math.hypot(vertices[i + 0], vertices[i + 1], vertices[i + 2]);
+        const v1 = vertices[i + 0];
+        const v2 = vertices[i + 1];
+        const v3 = vertices[i + 2];
+        const d = v1 * v1 + v2 * v2 + v3 * v3;
+        if (d < 1e-6) continue;
```

Comment:
In theory, this should never happen — vertices are never close to (0,0,0) point. If it happens, there must be a bug somewhere...